src/scripts/components/search.js

```diff
@@ -1,436 +1,426 @@
-                              current_page === page ? 'add' : 'remove'
-                          ]('active');
-                      }
-
-                      // scroll to top only if there is no hash
-                      if (!window.location.hash) {
-                          $('html, body').scrollTop(0);
-                      }
-                  }
-
-                  // init page nums
-                  initPageNums();
-
-                  // set initial page
-                  const searchParams = new URLSearchParams(
-                      window.location.search
-                  );
-                  if (searchParams.get('p') !== null)
-                      current_page = parseInt(searchParams.get('p'));
-                      window.history.replaceState({ page: current_page }, '', '');
-                  changePage(current_page);
-              }
-          } else {
-              const content = document.getElementsByClassName(
-                  'content'
-              )[0];
-              if (content) {
-                  content.innerHTML = '0 results';
-              }
-          }
-      }
-  );
-}
\ No newline at end of file
+const getAlgoliaConfig = () => {
+    const siteEnv = document.querySelector('html').dataset.env;
+    return configDocs[siteEnv].algoliaConfig;
+}
+
+const initializeAlgoliaIndex = () => {
+    const { index, appId, apiKey } = getAlgoliaConfig();
+    const client = algoliasearch(appId, apiKey);
+    return client.initIndex(index);
+}
+
+const getSiteLang = () => {
+    return document.querySelector('html').lang;
+}
+
+const getTitle = (hit) => {
+    let title = '';
+    title = hit['hierarchy']['lvl0'];
+    if (hit['hierarchy'].hasOwnProperty('lvl1')) {
+        if (hit['hierarchy']['lvl1'] !== null)
+            title += ` &raquo; ${hit['hierarchy']['lvl1']}`;
+    }
+    if (hit['hierarchy'].hasOwnProperty('lvl2')) {
+        if (hit['hierarchy']['lvl2'] !== null)
+            title += ` &raquo; ${hit['hierarchy']['lvl2']}`;
+    }
+    if (hit['hierarchy'].hasOwnProperty('lvl3')) {
+        if (hit['hierarchy']['lvl3'] !== null)
+            title += ` &raquo; ${hit['hierarchy']['lvl3']}`;
+    }
+
+    return title;
+}
+
+const appendSearchResultCountString = (count) => {
+    const tipueSearchResultCountDiv = document.createElement('div');
+    tipueSearchResultCountDiv.setAttribute('id', 'tipue_search_results_count');
+    tipueSearchResultCountDiv.innerText = `${count} results`;
+    const tipueSearchContentElement = document.getElementById('tipue_search_content');
+    tipueSearchContentElement.prepend(tipueSearchResultCountDiv);
+}
+
+const renderResults = (query, hits) => {
+    $('#tipue_search_input').val(query);
+    appendSearchResultCountString(hits.length);
+    let formatted_results = '';
+
+    if (hits.length) {
+        for (const i in hits) {
+            const hit = hits[i];
+            formatted_results += '<div class="hit">';
```

Comment:
actual rendering of results is legacy code, i have a Jira ticket for it here https://datadoghq.atlassian.net/browse/WEB-1928.    will address this in a later PR.